model/src/test/kotlin/DependencyGraphTest.kt

```diff
@@ -242,7 +242,7 @@ private fun id(group: String, artifact: String, version: String): Identifier =
  */
 private fun scopeDependencies(scopes: Set<Scope>, name: String): String = buildString {
     scopes.find { it.name == name }?.let { scope ->
-        scope.dependencies.forEach { dumpDependencies(it) }
+        scope.dependencies.sorted().forEach { dumpDependencies(it) }
```

Comment:
Or alternatively, stop `PackageReference` to implement `Comparable` and use matchers that assert equality of the data classes?

Reply:
> This relies on PackageReference to implement Comparable, which might change in the future. Can we instead use order-independent matchers in the callers?

I believe as long as `PackageReference` implements `Comparable` it is ok to make use of it. When later on turning `PackageReference` into non-comparable, it's impossible to miss this because the test will fail.

> Or alternatively, stop PackageReference to implement Comparable and use matchers that assert equality of the data classes?

I prefer to keep this out of scope of this PR, because it is already complex and large enough IMO.
Ditching comparable from `PackageReference` and `Scope` is not trivial.
 
I prefer to keep it as is, and have a dedicated PR later on which addresses dropping `Comparable`. Ok?

Reply:
> Can we agree on solving `Comparable` not implementing it later on?

Ok.